### DISCUSSION TITLE
Why is there both `qudt:PhysicalConstant` and `qudt:ConstantValue`

### DISCUSSION THREAD

**Discussion by dr-shorthair** [qudt/qudt-public-repo#419] [Q&A] [answered]
I don't see why there are two classes? 
In the [vocabulary of constants](http://www.qudt.org/doc/DOC_VOCAB-CONSTANTS.html) it seems to add an unnecessary additional indirection which makes it almost impossible to see the actual value in the user interface. 

Else also list the `qudt:ConstantValue` individuals somehow.

**Comment by steveraysteveray** [comment, marked as answer]
For an illustrative example of why there are two classes, take a look at [constant:SpeedOfLight_Vacuum](http://qudt.org/vocab/constant/SpeedOfLight_Vacuum). Notice that it has **_two_** linked values:

- [constant:Value_SpeedOfLight_Vacuum](http://qudt.org/vocab/constant/Value_SpeedOfLight_Vacuum)
- [constant:Value_SpeedOfLight_Vacuum_Imperial](http://qudt.org/vocab/constant/Value_SpeedOfLight_Vacuum_Imperial)

It is because a Constant (like any Quantity) can have values that exist as multiple sets of unit/value pairs. The **_concept_** of the speed of light is distinct from the unit/value pair you might use to express it.

Regarding the convenience of seeing the values, you are implying people would be routinely using our web pages to see them, is that correct? You do have alternatives:

- [The EDG interface](http://www.qudt.org/edg/tbl)
- Any application you might build that imports the QUDT ontologies, in which you could display them at the same time.

**Comment by dr-shorthair** [comment]
Thanks @steveraysteveray that is a helpful explanation. 

I generally send people to the web-pages for their initial encounter with QUDT. 
However, it doesn't work in this case because you can't (easily) click-through to the values. 
So I would suggest considering how to augment the browsing experience so that the values can be found more easily. 

The other criticism I have of the web-pages is that the pop-up does not actually _display_ the URI for the entity. 
Perhaps that is just my bias, but the URI is the linked-data grounding so I think it would help to make that more prominent.

**Comment by steveraysteveray** [comment]
Noted. This approach would be to overload the behavior of the pop-up generation code for the PhysicalConstant class and have it also fetch the unit and value of the associated ConstantValue instance.

**Comment by dr-shorthair** [comment]
I will create two new issues and close this one then.

**Comment by jhodgesatmb** [comment]
It is a good idea to show the URI of anything that has one. For the constants I suspect that anyone looking at them would want to check the value.

Jack Hodges, Ph.D.
Arbor Studios
